### PR TITLE
SYS-3338 make nft manager storage items bounded

### PR DESCRIPTION
**Pull request by thadouk**
Converts the nft manager storage items bounded.
This includes:
- batch size
- external references
- royalties

Added migration truncating and updating data that exceeded the boundaries
Updated tests and benchmarks
Updated log library std dependencies

## Type of change/Merge

🚨What type of change is this PR? </br>
_Put an `x` in the boxes that apply_

- [x] Release 
  - [x] Increase versions 
  - [ ] Baseline tests passed  
  - Release type:
    - [ ] Major release 
    - [x] Minor release 
    - [ ] Patch release

### CODE REVIEW

**Review by ivan-cholakov — APPROVED** [review]

File: pallets/nft-manager/src/lib.rs [hunk 1]
```diff
@@ -260,19 +276,20 @@ pub mod pallet {
...
     pub type BatchInfoId<T: Config> =
         StorageMap<_, Blake2_128Concat, NftBatchId, NftInfoId, ValueQuery>;
 
+    /// TODO: convert the key to BoundedVec once a runtime with the storage migration is applied.
     /// A mapping between an ExternalRef and a flag to show that an NFT has used it
     #[pallet::storage]
     #[pallet::getter(fn is_external_ref_used)]
     pub type UsedExternalReferences<T: Config> =
-        StorageMap<_, Blake2_128Concat, Vec<u8>, bool, ValueQuery>;
+        StorageMap<_, Blake2_128Concat, WeakBoundedVec<u8, NftExternalRefBound>, bool, ValueQuery>;
```
Comment (nahuseyoum) [inline pallets/nft-manager/src/lib.rs:292]: Unless I am missing something, it looks like we have 2 slightly different types for external refs: 
 - When we store them as part of the nft they use BoundedVec
 - When we store then in the `UsedExternalReferences` storage they use WeakBoundedVec
 Is this intentional?

**Review by nahuseyoum — APPROVED** [review]

File: pallets/nft-manager/src/lib.rs [hunk 2]
```diff
@@ -1227,57 +1289,70 @@ impl<T: Config> InnerCallValidator for Pallet<T> {
...
-    pub fn migrate_to_batch_nft<T: Config>() -> frame_support::weights::Weight {
+    pub fn migrate_to_bounded_nft<T: Config>() -> frame_support::weights::Weight {
         sp_runtime::runtime_logger::RuntimeLogger::init();
         log::info!("ℹ️  Nft manager pallet data migration invoked");
 
-        NftInfos::<T>::translate::<OldNftInfo, _>(|_, p| Some(p.upgraded::<T>()));
+        let mut keys_need_update = Vec::<Vec<u8>>::new();
+
+        Nfts::<T>::translate::<OldNft<T::AccountId>, _>(|_, p| {
+            let bound: usize = <NftExternalRefBound as sp_core::Get<u32>>::get() as usize;
```
[the remark this answers is not in the document]
Comment (thadouk) [reply pallets/nft-manager/src/lib.rs:1327]: no it can be outside the loop

File: pallets/nft-manager/src/lib.rs (hunk 1 above)
Comment (thadouk) [reply pallets/nft-manager/src/lib.rs:292]: yes, the storage representation is the same and we use it to migrate data. the next release will rectify things and make everything a "BoundedVector"

File: pallets/nft-manager/src/lib.rs [hunk 3]
```diff
@@ -1227,57 +1289,70 @@ impl<T: Config> InnerCallValidator for Pallet<T> {
...
 
-        log::info!("ℹ️  Migrated NftInfo data successfully");
+        log::info!("ℹ️ Found {:?} out of bounds external ref. Migrating...", keys_need_update.len());
+
+        for external_ref in keys_need_update.iter() {
+            let unique_reference = WeakBoundedVec::<u8, NftExternalRefBound>::force_from(
+                external_ref.clone(),
+                Some("Weak bound exceeded. Expected when migrating data to bounded"),
+            );
+            let value = UsedExternalReferences::<T>::get(&unique_reference);
```
[the remark this answers is not in the document]
Comment (thadouk) [reply pallets/nft-manager/src/lib.rs:1341]: I think so, since it's the key. Do you have an alternative in mind?

**Review by alexpArtos — APPROVED** [review]

**Pull request merged**